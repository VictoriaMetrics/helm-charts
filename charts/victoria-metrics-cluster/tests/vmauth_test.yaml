suite: vmauth templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - vmauth-deployment.yaml
tests:
  - it: deployment should match empty snapshot
    asserts:
      - matchSnapshot: {}
  - it: deployment should match snapshot
    set:
      vmauth:
        enabled: true
    asserts:
      - matchSnapshot: {}
  - it: deployment should match snapshot with fullnameOverride, extraLabels and podLabels
    set:
      vmauth:
        enabled: true
        name: auth
        fullnameOverride: vmauth-node
        extraLabels:
          extraLabelName: extraLabelValue
        podLabels:
          podLabelName: podLabelValue
    asserts:
      - matchSnapshot: {}
