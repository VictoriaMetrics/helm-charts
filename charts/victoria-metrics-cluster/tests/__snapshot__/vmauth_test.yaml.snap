deployment should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: vmauth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-cluster
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-metrics-cluster-0.1.1
      name: RELEASE-NAME-victoria-metrics-cluster-vmauth
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          app: vmauth
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: victoria-metrics-cluster
      template:
        metadata:
          labels:
            app: vmauth
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: victoria-metrics-cluster
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --auth.config=/config/auth.yml
                - --envflag.enable=true
                - --envflag.prefix=VM_
                - --loggerFormat=json
              image: victoriametrics/vmauth:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                initialDelaySeconds: 5
                periodSeconds: 15
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vmauth
              ports:
                - containerPort: 8427
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 5
              volumeMounts:
                - mountPath: /config
                  name: config
          serviceAccountName: RELEASE-NAME-victoria-metrics-cluster
          volumes:
            - name: config
              secret:
                secretName: RELEASE-NAME-victoria-metrics-cluster-vmauth
deployment should match snapshot with fullnameOverride, extraLabels and podLabels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: auth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-cluster
        app.kubernetes.io/version: 0.1.0
        extraLabelName: extraLabelValue
        helm.sh/chart: victoria-metrics-cluster-0.1.1
      name: vmauth-node
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          app: auth
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: victoria-metrics-cluster
      template:
        metadata:
          labels:
            app: auth
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: victoria-metrics-cluster
            podLabelName: podLabelValue
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --auth.config=/config/auth.yml
                - --envflag.enable=true
                - --envflag.prefix=VM_
                - --loggerFormat=json
              image: victoriametrics/vmauth:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                initialDelaySeconds: 5
                periodSeconds: 15
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vmauth
              ports:
                - containerPort: 8427
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 5
              volumeMounts:
                - mountPath: /config
                  name: config
          serviceAccountName: RELEASE-NAME-victoria-metrics-cluster
          volumes:
            - name: config
              secret:
                secretName: vmauth-node
